core/src/main/java/org/apache/carbondata/core/datastore/chunk/store/impl/LocalDictDimensionDataChunkStore.java

```diff
@@ -61,10 +61,7 @@ public void fillVector(int[] invertedIndex, int[] invertedIndexReverse, byte[] d
     int columnValueSize = dimensionDataChunkStore.getColumnValueSize();
     int rowsNum = data.length / columnValueSize;
     CarbonColumnVector vector = vectorInfo.vector;
-    if (!dictionary.isDictionaryUsed()) {
-      vector.setDictionary(dictionary);
-      dictionary.setDictionaryUsed();
-    }
+    vector.setDictionary(dictionary);
```

Comment:
Is the same handling required in `fillRow` method also in the same class?..If required then `isDictionaryUsed` and `setDictionaryUsed` API's will not be required and those can also be removed from the interface

Reply:
I have checked this while coding, fill row is our method. So, no issues. 
Only this vector is spark vector and dictionary needs to clear for it.

Reply:
Both the method are called in carbon flow for vector filling. One is direct fill case and the other one is old vector fill flow. Please cross check once